### PR TITLE
Update to new ESH feature

### PR DESCRIPTION
**Pull request by dvanherbergen**
In preparation of https://github.com/openhab/openhab-distro/issues/11

Signed-off-by: Davy Vanherbergen <davy.vanherbergen@gmail.com>

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -163,6 +163,11 @@
             <url>https://jcenter.bintray.com/</url>
         </repository>
         <repository>
+            <id>shk-bintray</id>
+            <name>JCenter Repository</name>
```
Comment (maggu2810) [inline pom.xml]: perhaps you should choose another name (or none).

File: features/karaf/src/main/feature/feature.xml [hunk 2]
```diff
@@ -12,8 +12,8 @@
 <features name="${project.artifactId}-${project.version}" xmlns="http://karaf.apache.org/xmlns/features/v1.0.0">
 
     <feature name="openhab-runtime-base" description="openHAB Runtime Base" version="${project.version}">
-        <feature>shk-esh-base</feature>
-        <feature>shk-esh-io-console-karaf</feature>
+        <feature>esh-base</feature>
```
Comment (maggu2810) [inline features/karaf/src/main/feature/feature.xml]: As esh-base has been stripped off, you will perhaps add some other features, e.g.
```text
    <feature>esh-base</feature>
    <feature>esh-io-console-karaf</feature>
    <feature>esh-io-rest-sitemap</feature>
    <feature>esh-model-item-runtime</feature>
    <feature>esh-model-persistence-runtime</feature>
    <feature>esh-model-rule-runtime</feature>
    <feature>esh-model-script-runtime</feature>
    <feature>esh-model-sitemap-runtime</feature>
    <feature>esh-model-thing-runtime</feature>
    <feature>esh-storage-mapdb</feature>
    <feature>esh-ui-iconset-classic</feature>
```

File: pom.xml (hunk 1 above)
Comment (dvanherbergen) [inline pom.xml]: How about "Bintray Repository for shk" ?

Comment (maggu2810) [inline pom.xml]: Okay for me.

File: features/karaf/src/main/feature/feature.xml (hunk 2 above)
Comment (dvanherbergen) [inline features/karaf/src/main/feature/feature.xml]: Those have been added to the features where needed. I didn't add them to the runtime base yet to keep it lean.

**Comment by dvanherbergen** [comment]
Name was updated.

File: features/karaf/src/main/feature/feature.xml (hunk 2 above)
Comment (maggu2810) [inline features/karaf/src/main/feature/feature.xml]: > Those have been added to the features where needed.

That is good and correct for the direct dependencies .

Hm, I could be wrong, but there is IMHO stuff that no one depend but should be part of the base runtime (e.g. storage-mapdb).
I would also assume that all the model stuff are seen as part of the openhab-runtime-base feature and so installed every time (regardless if there is something that needs it).

Comment (dvanherbergen) [inline features/karaf/src/main/feature/feature.xml]: I left it out as I was thinking on reusing the openhab-runtime-base for the minimal distro, but we'll probably need to do something special there anyway.
I will add the features as you suggested above. Any others that should be included?

**Comment by dvanherbergen** [comment]
Suggested features added.

**Comment by maggu2810** [comment]
> I left it out as I was thinking on reusing the openhab-runtime-base for the minimal distro, but we'll probably need to do something special there anyway.
I will add the features as you suggested above. Any others that should be included?

We will have to test it, to see if there is other stuff missing. But I believe all is fine.

I would suggest using "openhab-runtime-base" for a base of all openhab-runtimes.
And create one for each distribution that depend on openhab-runtime-base and add perhaps some other features or bundles (openhab-runtime-offline, openhab-runtime-online [will perhaps be include openhab-runtime-offline only], openhab-runtime-minimal, ...)
But I think that could be done in a PR for the minimal dist or some other.

All fine for me.

**Comment by maggu2810** [comment]
I cannot merge it, not the necessary permissions...

**Comment by kaikreuzer** [comment]
Thanks! FTR, I have just build a new stable ESH release, so that the ESH features are now available as snapshots in repo.eclipse.org.
Now we just have to get the Cloudbees builds work again after they ran into quota issues...

**Comment by maggu2810** [comment]
Do you plan to create a 0.8.0.b2?

**Comment by kaikreuzer** [comment]
I knew you were asking for it :-)
Yes, once we see that it all works nicely, we can do a 0.8.0.b2. I leave it up to you when you think is the best moment for it (considering that there is a long list of PRs open)

**Comment by dvanherbergen** [comment]
> Now we just have to get the Cloudbees builds work again after they ran into quota issues... 

Can we not drop the distribution and products projects in openhab2? That should save some MB's.

**Comment by kaikreuzer** [comment]
I plan to drop it as soon as the Karaf distro is the official one - at the moment, people are still using the distro built in openhab/openhab2.
I planned a 2.0 beta release next Monday, together with the openHAB 1.8 release - that's when I want to make openhab/openhab-distro the official one for all further development (even for openHAB 1 addons).
That's why https://github.com/openhab/openhab-distro/issues/10 has such a high prio for me - it must simply be working smoothly and stable by next Monday!

**Comment by dvanherbergen** [comment]
> it must simply be working smoothly and stable by next Monday!

Right. No pressure :wink: 

I guess you could reduce the build history of https://openhab.ci.cloudbees.com/job/openHAB/ to no history. Other than that, I don't immediately see any large areas for improvement.

**Pull request merged**